marklogic-data-hub/src/main/java/com/marklogic/hub/deploy/commands/LoadUserArtifactsCommand.java

```diff
@@ -188,44 +201,45 @@ public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) th
                     }
                 });
             }
-
-            // let's do step-definitions
-            if (stepDefPath.toFile().exists()) {
-                Files.walkFileTree(stepDefPath, new SimpleFileVisitor<Path>() {
-                    @Override
-                    public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
-                        executeWalk(
-                            dir,
-                            stepDefModulesFinder,
-                            propertiesModuleManager,
-                            stepResourceToURI,
-                            buildMetadata(hubConfig.getStepDefinitionPermissions(), "http://marklogic.com/data-hub/step-definition"),
-                            stagingStepDefDocumentWriteSet,
-                            finalStepDefDocumentWriteSet
-                        );
-                        return FileVisitResult.CONTINUE;
-                    }
-                });
-            }
-
-
-            // let's do flows
-            if (flowPath.toFile().exists()) {
-                Files.walkFileTree(flowPath, new SimpleFileVisitor<Path>() {
-                    @Override
-                    public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
-                        executeWalk(
-                            dir,
-                            flowDefModulesFinder,
-                            propertiesModuleManager,
-                            flowResourceToURI,
-                            buildMetadata(hubConfig.getFlowPermissions(), "http://marklogic.com/data-hub/flow"),
-                            stagingFlowDocumentWriteSet,
-                            finalFlowDocumentWriteSet
-                        );
-                        return FileVisitResult.CONTINUE;
-                    }
-                });
+            ArtifactManager artifactManager = new ArtifactManagerImpl(hubConfig);
+            List<String> roles = new LinkedList<>();
+            RolesService.on(stagingClient).getRoles().elements().forEachRemaining(
+                (roleNode) -> roles.add(roleNode.asText())
+            );
+            ArtifactService artifactService = ArtifactService.on(stagingClient);
+            ObjectMapper mapper = new ObjectMapper();
+            // New way to deploy Artifacts via Data Services
+            for (ArtifactTypeInfo typeInfo: artifactManager.getArtifactTypeInfoList()) {
+                final String artifactType = typeInfo.getType();
+                final Path artifactPath = hubConfig.getHubProject().getProjectDir().toAbsolutePath().resolve(typeInfo.getDirectory().replaceAll("^/","")).toAbsolutePath();
```

Comment:
I think this is a good method to add to HubProject. I think HubProject should be aware of an ArtifactTypeInfo, and thus have a method of:

    Path getArtifactTypePath(ArtifactTypeInfo)

Reply:
Moved the logic for the path into the HubProject

Reply:
Moved the logic for the path into the HubProject

Reply:
Moved the logic for the path into the HubProject